doc/specs/spec-template.md

```diff
@@ -0,0 +1,50 @@
+---
+author: <first-name> <last-name> <github-id>/<email>
+created on: <yyyy-mm-dd>
+last updated: <yyyy-mm-dd>
+issue id: <github issue id>
+---
+
+# Spec Title
+
+## Abstract
+
+[comment]: # Outline what this spec describes
+
+## Inspiration
+
+[comment]: # What were the drivers/inspiration behind the creation of this spec.
+
+## Solution Design
+
+[comment]: # Outline the design of the solution. Feel free to include ASCII-art diagrams, etc.
+
+## UI/UX Design
+
+[comment]: # What will this fix/feature look like? How will it affect the end user?
+
+## Capabilities
+
+[comment]: # Discuss how the proposed fixes/features impact the following key consdierations:
+
+### Security
+
+### Reliability
+
+### Performance, Power, and Efficiency
+
+### Accessibility

```

Comment:
Compatibility too? At least to spark a justification on why something is OK to be a breaking change (or not) versus a previous version?